server/queue/fifo.go

```diff
@@ -198,23 +199,50 @@ func (q *fifo) Extend(_ context.Context, agentID int64, taskID string) error {
 }
 
 // Info returns internal queue information.
-func (q *fifo) Info(_ context.Context) InfoT {
+func (q *fifo) Info(c context.Context) InfoT {
 	q.Lock()
 	stats := InfoT{}
 	stats.Stats.Workers = len(q.workers)
 	stats.Stats.Pending = q.pending.Len()
 	stats.Stats.WaitingOnDeps = q.waitingOnDeps.Len()
 	stats.Stats.Running = len(q.running)
 
+	// Create a map of agent IDs to names
+	agentNames := make(map[int64]string)
```

Comment:
Would suggest to query the agent endpoint for the names using the api instead for normalized data structures.